index.js

```diff
@@ -123,18 +127,16 @@ function connect (appKey, accessToken, _opts) {
     next()
   }
 
-  function end (done) {
+  function end(done) {
     ended = true
-
-    if (request) {
-      request.end()
-      request.on('end', done)
-    } else {
-      done()
-    }
+    if (request) request.destroy()
+    if (responseStream) responseStream.destroy()
```

Comment:
this is the big fix here, previously we were calling to end the `request` stream (and also listening for its "end" event, which never actually triggered), but _not_ actually ending the stream that was piping through to the returned stream, so it wouldn't ever properly close.